### PR TITLE
Galactic HiPS & Basic HiPS properties support

### PR DESCRIPTION
**Pull request by imbasimba**

### CODE REVIEW

File: engine/wwtlib/HipsProperties.cs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+namespace wwtlib
+{
+
+    public class HipsProperties
+    {
+        public Dictionary<string, string> Properties = new Dictionary<string, string>();
+        public VoTable VoTable = null;
+        public bool downloadComplete = false;
+        private WebFile webFile;
+        public static HipsProperties GetPropertiesCat(string url)
```
Comment (pkgw) [inline engine/wwtlib/HipsProperties.cs:13]: Is this function used at all? I don't think so.

File: engine/wwtlib/HipsProperties.cs [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            webFile = new WebFile(propsUrl);
+            webFile.OnStateChange = LoadData;
+            webFile.Send();
+        }
+
+        private void LoadData()
+        {
+            if (webFile.State == StateType.Error)
+            {
+                Script.Literal("alert({0})", webFile.Message);
```
Comment (pkgw) [inline engine/wwtlib/HipsProperties.cs:116]: We shouldn't be alert()ing.

File: engine/wwtlib/HipsProperties.cs [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+            }
+            if (!props.Properties.ContainsKey("dataproduct_type"))
+            {
+                return false;
+            }
+            if (props.Properties["dataproduct_type"] == "cube")
+            {
+                return false;
+            }
+            if (props.Properties.ContainsKey("client_category") && props.Properties["client_category"].ToLowerCase().IndexOf("solar") > -1)
```
Comment (pkgw) [inline engine/wwtlib/HipsProperties.cs:174]: Do you know what this test is all about? It seems odd to me.

**Review by pkgw — COMMENTED** [review]
Mostly looking good, but I am concerned about the `alert()` in the properties code.

File: engine/wwtlib/HipsProperties.cs (hunk 1 above)
Comment (imbasimba) [reply engine/wwtlib/HipsProperties.cs:13]: I thought I was going to need it for Catalog HiPS, but that turned out to be false. I'll remove it

File: engine/wwtlib/HipsProperties.cs (hunk 2 above)
Comment (imbasimba) [reply engine/wwtlib/HipsProperties.cs:116]: Agreed

File: engine/wwtlib/HipsProperties.cs (hunk 3 above)
Comment (imbasimba) [reply engine/wwtlib/HipsProperties.cs:174]: Hmm... Sure looks unnecessary to me. The function itself is not yet used, but I'll have to take a closer look when it is time to fully activate this part.

**Pull request merged**